zerver/tests/test_invite.py

```diff
@@ -997,6 +999,51 @@ def test_multi_user_invite(self) -> None:
             ]
         )
 
+    def test_direct_notification_for_accepted_invitation(self) -> None:
+        """
+        New test to check if notification-bot message is sent to the inviter
+        when send_notification is false.
+        """
+        self.login("hamlet")
+
+        invitee = self.nonreg_email("alice")
+        self.assert_json_success(self.invite(invitee, ["Denmark"], False))
+        self.assertTrue(find_key_by_email(invitee))
+        self.submit_reg_form_for_user(invitee, "password")
+
+        invitee_profile = self.nonreg_user("alice")
+        self.assertFalse(invitee_profile.is_realm_admin)
+
+        # Check for absence of inviter notification message
+        self.assertFalse(
+            any(
+                msg.content.startswith(f"@_**alice_zulip.com|{invitee_profile.id}** accepted your")
+                for msg in Message.objects.all()
+            ),
+            "Unexpected inviter notification message found",
```

Comment:
I think a better way would be to just find the last message and then just check whether the content and sender are as expected.